app/(tabs)/clubs.tsx

```diff
@@ -0,0 +1,142 @@
+import { View, StyleSheet, Text, ScrollView, TouchableOpacity, Dimensions } from 'react-native';
+import React, { useState } from 'react';
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
+import { useRouter } from "expo-router";
+import FontAwesome from '@expo/vector-icons/FontAwesome';
+import { useFocusEffect } from '@react-navigation/native';
+import { Input, InputField, InputIcon, InputSlot } from '@/components/ui/input';
+import { useQuery } from '@tanstack/react-query';
+import { fetchClubByName } from '../../api/clubService';
+import CardLayout from '@/components/cards/CardLayout';
+import { SimplifiedClubInfo } from '../../types/ClubInfoTypes';
+
+const { width: screenWidth } = Dimensions.get('window');
+
+type FormData = {
+  searchInput: string;
+};
+
+export default function Clubs() {
+  const router = useRouter();
+  const { control, handleSubmit, formState: { errors }, reset } = useForm<FormData>();
+  const [inputText, setInputText] = useState('');
+  const { data: clubs, isLoading, isError } = useQuery<[SimplifiedClubInfo], Error>(
+    {
+      queryKey: ['clubSearch', inputText],
+      queryFn: () => fetchClubByName(inputText),
+      enabled: !!inputText,
+    }
+  );
+
+  // Clears form state when screen gains focus
+  useFocusEffect(
+    React.useCallback(() => {
+      reset();
+      setInputText('');
+    }, [reset])
+  );
+
+  const handleSearchClick: SubmitHandler<FormData> = (data) => {
```

Comment:
You should probably add a debounce on the input to prevent unecessary network requests.